lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -942,9 +939,9 @@ def create_autoscaling_group(connection, module):
                     attach_lb_target_groups(connection, group_name, tgs_to_attach)
 
         # check for attributes that aren't required for updating an existing ASG
-        desired_capacity = desired_capacity or as_group['DesiredCapacity']
-        min_size = min_size or as_group['MinSize']
-        max_size = max_size or as_group['MaxSize']
+        desired_capacity = module.params.get('desired_capacity', as_group['DesiredCapacity'])
```

Comment:
This doesn't work as expected, unfortunately.
```
>>> test_dict = {'one': 1, 'two': None}
>>> print(test_dict.get('two'))
None
>>> print(test_dict.get('two', 2))
None
>>> print(test_dict.get('three', 3))
3
```

Reply:
```
The full traceback is:
Traceback (most recent call last):
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_ypla3C/ansible_module_ec2_asg.py", line 1353, in <module>
    main()
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_ypla3C/ansible_module_ec2_asg.py", line 1342, in main
    create_changed, asg_properties = create_autoscaling_group(connection, module)
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_ypla3C/ansible_module_ec2_asg.py", line 967, in create_autoscaling_group
    update_asg(connection, **ag)
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_ypla3C/ansible_modlib.zip/ansible/module_utils/cloud.py", line 153, in retry_func
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid type for parameter DesiredCapacity, value: None, type: <type 'NoneType'>, valid types: <type 'int'>, <type 'long'>
```
Even though this is prettier, I think we might need to do what you initially did by checking if not None:
`desired_capacity = desired_capacity if desired_capacity is not None else as_group['DesiredCapacity']` etc.